pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/onchistory/OncHistoryUploadService.java

```diff
@@ -125,17 +130,33 @@ public void upload(String fileContent) {
     /**
      * Given participant short IDs in uploaded rows, verify the short ID, and get and record associated
      * participant IDs and medical record IDs
+     * verify if any exited participant(s) exists in passed onc history records.
      *
-     * @throws OncHistoryValidationException for failed verification
+     * @throws OncHistoryValidationException for failed verifications
      */
     protected Map<Integer, Integer> getParticipantIds(List<OncHistoryRecord> oncHistoryRecords,
                                                       ParticipantIdProvider participantIdProvider, boolean updateElastic) {
         Map<Integer, Integer> medIds = new HashMap<>();
+        List<String> exitedParticipants = new ArrayList<>();
 
         ParticipantDao participantDao = ParticipantDao.of();
 
+        String esIndex = new DDPInstanceDao().getDDPInstanceByInstanceName(realm).orElseThrow().getEsParticipantIndex();
         for (OncHistoryRecord rec : oncHistoryRecords) {
             int participantId = participantIdProvider.getParticipantIdForShortId(rec.getParticipantTextId());
+            Optional<ElasticSearchParticipantDto> ptpData = elasticSearchService.getParticipantDocumentByShortId(
+                    rec.getParticipantTextId(), esIndex);
+            if (ptpData.isEmpty()) {
+                throw new OncHistoryValidationException("Invalid short ID " + rec.getParticipantTextId());
+            }
+            if (ptpData.get().getStatus().isPresent() && ptpData.get().getStatus().get().startsWith("EXITED")) {
+                exitedParticipants.add(rec.getParticipantTextId());
+                log.info("Participant {} is exited, skipping", rec.getParticipantTextId()); //todo
+            }
+
+            if (!exitedParticipants.isEmpty()) {
+                continue; //skip MR verify/creation. continuing to collect any other exited participant hruids.
```

Comment:
Comments at the end of a line of code do not meet the DSM style guide.
Are you getting a checkstyle warning?